### PR TITLE
address broken build and json error related to escape characters while retrieving from neo4j

### PR DESCRIPTION
**Pull request by ramach**
this patch is to address broken build. temporarily changing to postgres as default. also addresses escape character in json while using neo4j

### CODE REVIEW

File: plugins/hive-plugin/src/main/resources/ground.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 edu.berkeley.ground.model.config.dbClient=test_connection
-edu.berkeley.ground.model.config.factoryType=neo4j
+edu.berkeley.ground.model.config.factoryType=postgres
```
Comment (vsreekanti) [inline plugins/hive-plugin/src/main/resources/ground.properties]: Can we switch this back to neo4j?

**Comment by ramach** [comment]
Like I said changing to neo4j default breaks tests - which needs some debugging. Am not sure if the problem is in plugin, neo4j interface or neo4j itself

the error is strange - since it is looking for a relationship which does not make sense

Tests in error: 
  testDatabaseOps(edu.berkeley.ground.plugins.hive.GroundMetastoreTest): Parent Structures.testgroundstoredb1 is not in Item Structures.testgroundstoredb2

though all nodes & edges are created as expected

**Comment by vsreekanti** [comment]
Understood. It would be really great if we could get this resolved ASAP since, like we've talked about, we really would like to be able to prescribe Neo4j as the default datastore for Ground. It would be a bummer if we would have to change that because of this bug. 

Let me know if there's anyway I can help with this.

**Comment by ramach** [comment]
I am looking in to this and hope to resolve today. It is easy to test - just change "factoryType" to neo4j and run this test GroundMetastoreTest using mvn

Any additional insights/leads will help. Will update (am debugging now)

**Comment by ramach** [comment]
I got some more details. WIth neo4j factory,  while creating a brand new database nodeversion it is looking for a relationship (a parent) which is completely a different database - so no such update should be attempted (see below)

Still have not nailed if the error is in plugin or neo4j factory - thought I will share this detail

This does not happen with postgres

.
10:33:16.463 [main] INFO edu.berkeley.ground.api.models.neo4j.Neo4jEdgeVersionFactory - Created edge version e4d7e6bcdfa69376a0cf04dd861eb in edge Edges.Nodes.testgroundstoredb1.
10:33:16.514 [main] DEBUG edu.berkeley.ground.plugins.hive.GroundDatabase - Fetching database node: testgroundstoredb2
10:33:16.521 [main] DEBUG edu.berkeley.ground.plugins.hive.GroundDatabase - Not found - Creating databsae node: testgroundstoredb2
10:33:16.563 [main] INFO edu.berkeley.ground.api.models.neo4j.Neo4jNodeFactory - Created node testgroundstoredb2.
10:33:16.569 [main] INFO edu.berkeley.ground.api.models.neo4j.Neo4jStructureFactory - Created structure testgroundstoredb2.
10:33:16.569 [main] DEBUG edu.berkeley.ground.plugins.hive.GroundDatabase - node structure created edu.berkeley.ground.api.models.Structure@3d4818e8
10:33:16.608 [main] INFO edu.berkeley.ground.api.models.neo4j.Neo4jStructureFactory - Created structure testgroundstoredb2.
10:33:16.608 [main] INFO edu.berkeley.ground.plugins.hive.GroundDatabase - Node and Structure Nodes.testgroundstoredb2, Structures.testgroundstoredb2
10:33:16.622 [main] ERROR edu.berkeley.ground.plugins.hive.GroundDatabase - Failure to create a database node: {}
edu.berkeley.ground.exceptions.GroundException: Parent Structures.testgroundstoredb1 is not in Item Structures.testgroundstoredb2.
    at edu.berkeley.ground.api.versions.neo4j.Neo4jItemFactory.update(Neo4jItemFactory.java:58)
    at edu.berkeley.ground.api.models.neo4j.Neo4jStructureFactory.update(Neo4jStructureFactory.java:106)
    at edu.berkeley.ground.api.models.neo4j.Neo4jStructureVersionFactory.create(Neo4jStructureVersionFactory.java:65)
    at edu.berkeley.ground.plugins.hive.GroundDatabase.createDatabase(GroundDatabase.java:150)
    at edu.berkeley.ground.plugins.hive.GroundMetaStore.createDatabase(GroundMetaStore.java:195)

**Comment by ramach** [comment]
I think I found the problem. Will revise the patch and update soon.

**Comment by ramach** [comment]
I updated the fork after restoring neo4j as default. Partition test is disabled temporarily as we run out of resources in neo4j

the problem i found is while creating a StructureVersion using a parent, the list EMPTY_PARENT_LIST is not really empty. It contains entries corresponding to an earlier DB that was created as part of testing. Now am using a fresh list - we need a closer look and update.

**Comment by vsreekanti** [comment]
Great. What's the issue that we're getting with the partition test? Could it be a similar issue?

**Comment by ramach** [comment]
partition test works now. Last commit addressed that as well.

let me know if it fails. Only issue I faced is running out neo4j
connections.

**Comment by vsreekanti** [comment]
Hm. That sounds like it could be because we're not closing connections somewhere. What exactly is the error?

**Comment by ramach** [comment]
Each test runs fine when ran 2 at a time (Ignore one and run any other 2 at
once)

If I run all of them together see this error,

Failed to acquire a session with Neo4j as all the connections in the
connection pool are already occupied by other sessions. Please close unused
session and retry. Current Pool size: 50. If your application requires
running more sessions concurrently than the current pool size, you should
create a driver with a larger connection pool size.
K

**Comment by vsreekanti** [comment]
I just pushed a bugfix. The `Neo4jStructureVersionFactory` was leaving connections open, and I'm assuming you're retrieving `StructureVersion`s pretty often. That should fix this bug. Please merge and let me know.

**Comment by ramach** [comment]
that helped (yes we access structure version many times). Now all tests
pass - pasting last few lines of the build.

I pushed changes to the fork. Please review and commit when you get a chance

regards
Krishna

21:29:36.713 [main] INFO
edu.berkeley.ground.api.models.neo4j.Neo4jStructureVersionFactory -
Retrieved structure version 95a8afdaffda12cbb3a4f7098b6c4b3 in structure
Structures.Nodes.testgroundstoredbpart1.

21:29:36.735 [main] INFO
edu.berkeley.ground.api.models.neo4j.Neo4jEdgeVersionFactory - Created edge
version dfe8ba885d885aa0eabe87ec in edge
Edges.Nodes.testgroundstoredb1withtable.

21:29:36.749 [main] INFO
edu.berkeley.ground.api.models.neo4j.Neo4jNodeVersionFactory - Retrieved
node version 8d983f6de3f79afc379ba2b09e8 in node
Nodes.testgroundstoredbpart1:partitiontesttable1:US:CA.

21:29:36.761 [main] INFO
edu.berkeley.ground.api.models.neo4j.Neo4jNodeVersionFactory - Retrieved
node version b7fdd8aafef4dc0e7c0d3b1dabbfa6cda481 in node
Nodes.testgroundstoredbpart1:partitiontesttable1:US:MA.

Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 5.062 sec

Results :

Tests run: 3, Failures: 0, Errors: 0, Skipped: 0

[INFO]

[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ ground-hive-plugin ---

[INFO] Building jar:
/Users/krishnar/forked-ground/ground/plugins/hive-plugin/target/ground-hive-plugin-0.1-SNAPSHOT.jar

[INFO]

## [INFO]

[INFO] Building Ground 0.1-SNAPSHOT

## [INFO]

[INFO]

[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ ground ---

[INFO]

[INFO] --- jacoco-maven-plugin:0.7.6.201602180812:prepare-agent
(prepare-agent) @ ground ---

[INFO] argLine set to
-javaagent:/Users/krishnar/.m2/repository/org/jacoco/org.jacoco.agent/0.7.6.201602180812/org.jacoco.agent-0.7.6.201602180812-runtime.jar=destfile=/Users/krishnar/forked-ground/ground/target/jacoco.exec

## [INFO]

[INFO] Reactor Summary:

[INFO]

[INFO] Ground Core ........................................ SUCCESS [
13.762 s]

[INFO] Ground Ingest Plugin ............................... SUCCESS [
5.285 s]

[INFO] Ground Hive Plugin ................................. SUCCESS [
7.083 s]

[INFO] Ground ............................................. SUCCESS [
0.002 s]

## [INFO]

[INFO] BUILD SUCCESS

## [INFO]

**Pull request merged**